fala/apps/adviser/tests/test_crown_dependencies_view.py

```diff
@@ -8,46 +8,42 @@ class PostcodeValidationTest(SimpleTestCase):
     client = Client()
     url = reverse("search")
 
-    def test_guernsey_postcode(self):
-        data = {"postcode": "GY1"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "The postcode GY1 is in Guernsey")
-
-    def test_invalid_guernsey_postcode(self):
-        data = {"postcode": "GY1"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "Legal Aid in Guernsey")
-
-    def test_invalid_scottish_postcode(self):
-        data = {"postcode": "AB"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "The postcode AB is in Scotland")
-
-    def test_valid_scottish_postcode(self):
-        data = {"postcode": "AB11 5BN"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "The postcode AB11 5BN is in Scotland")
-
-    def dont_test_valid_english_postcode(self):
-        data = {"postcode": "M2 3WQ"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "in order of closeness to")
+    def test_region_postcodes(self):
+        test_cases = [
+            {"postcode": "GY1", "message": "The postcode GY1 is in Guernsey"},
+            {"postcode": "BT93 8AD", "message": "The postcode BT93 8AD is in Northern Ireland"},
+            {"postcode": "IM4", "message": "The postcode IM4 is in the Isle of Man"},
+            {"postcode": "JE2 3FN", "message": "The postcode JE2 3FN is in Jersey"},
+            # English Postcode
+            {"postcode": "M2 3WQ", "message": "in order of closeness to"},
+            # Scottish Postcode
+            {"postcode": "AB11 5BN", "message": "These results cover England and Wales."},
+            # Lower case Postcode
+            {"postcode": "im4", "message": "The postcode IM4 is in the Isle of Man"},
+            # Invalid post code
```

Comment:
I'm not sure this test is actually for an invalid postcode, is that right? And the one below. It looks like from the message that they do a successful search. So the comment confuses me a bit. Could it be reworded to be clearer what sort of postcode input this is?

Reply:
Your right, I've added in a comment explaining the "invalid postcode" which is technically valid situation. 

It's because we're not adding any numbers after the first two characters. 

I've removed BT as that proves nothing now looking at this again with your comments. 

So AB will search for Scottish addresses allowing us to pick up the region, it's a bad postcode though as it wont tell you which part of Scotland, e.g. AB1, AB2 etc. However we still pick up that it's a Scottish postcode. It's still not great but there is another ticket in the pipeline to improve postcodes and their regions as not all Scottish postcodes are Scottish postcodes. Fun times.